### PR TITLE
Customer

### PR DESCRIPTION
**Pull request by Sevitas**
I've created package Impl for better organisation of implemented DAO classes,
? Address could be probably used for other entities instead of destination, we should also create ENUM for different types of addresses

### CODE REVIEW

**Comment by 445611** [comment]
Javadoc missing + checkstyle fix 
Hash + equals methods missing

File: src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    @NotNull
+    private String IdCardNumber;
+
+    @NotNull
+    private String email;
+
+    @NotNull
+    @OneToMany(mappedBy = "customer")
+    private Set<Address> addresses;
```
Comment (445611) [inline src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java]: there sould be used just:
private Address address; 
or not ?

File: src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    private String phoneNumber;
+
+    @NotNull
+    private String IdCardNumber;
+
+    @NotNull
+    private String email;
+
+    @NotNull
+    @OneToMany(mappedBy = "customer")
```
Comment (445611) [inline src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java]: why mappedBy customer ? 
dont you just miss copypaste that from attribute reservations?
You should be fine with @OneToOne

File: src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java (hunk 1 above)
Comment (Sevitas) [reply src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java]: customer can have many addresses, one were he lives, second billing

Comment (445611) [reply src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java]: do we really want to go into such details ?

Comment (Sevitas) [reply src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java]: we can ask on seminar

File: src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    public Date getBirthDate() {
+        return birthDate;
+    }
+
+    public void setBirthDate(Date birthDate) {
+        this.birthDate = birthDate;
+    }
+
+    public Set<Reservation> getReservations() {
+        return reservations;
```
Comment (445611) [inline src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java]: When returning collections , use Collections.unmodifiableSet

File: src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java (hunk 1 above)
Comment (445611) [reply src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java]: agree :)

Comment (Sevitas) [reply src/main/java/fi/muni/cz/pa165/travelagency/entity/Customer.java]: I fixed all other errors,
I have edited checkstyle for accepting expressions like "If(true) return false;"
I cannot push probably this pull request has to be handled first. (not sure)

**Comment by Sevitas** [comment]
fixed errors and uploaded new version,
check edited checkstyle.xml for changes, now it is possible to write "**if (true) return false**".

**Comment by Sevitas** [comment]
fixed errors, uploaded new version

**Pull request merged**